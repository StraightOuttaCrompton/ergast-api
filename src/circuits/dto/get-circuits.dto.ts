import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
