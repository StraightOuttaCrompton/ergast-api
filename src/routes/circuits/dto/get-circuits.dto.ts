import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
