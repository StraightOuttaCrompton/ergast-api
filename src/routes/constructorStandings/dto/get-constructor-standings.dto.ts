import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
