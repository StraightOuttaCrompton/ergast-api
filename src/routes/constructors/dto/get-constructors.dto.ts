import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number; // TODO: validate for actual status values

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    constructorStandings?: number;
}
