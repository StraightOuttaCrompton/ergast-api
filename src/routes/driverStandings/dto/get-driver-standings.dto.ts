import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriverStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
