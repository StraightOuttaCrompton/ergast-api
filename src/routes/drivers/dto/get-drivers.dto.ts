import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriversDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    driverStandings?: number;
}
