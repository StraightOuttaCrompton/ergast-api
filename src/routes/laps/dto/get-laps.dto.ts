import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetLapsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;
}
