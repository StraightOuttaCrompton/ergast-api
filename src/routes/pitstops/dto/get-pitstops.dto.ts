import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetPitstopsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    lapNumber: number;
}
