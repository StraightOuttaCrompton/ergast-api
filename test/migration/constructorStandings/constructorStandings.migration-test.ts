import querystring from "querystring";
import { GetConstructorStandingsDto } from "src/routes/constructorStandings/dto/get-constructor-standings.dto";
import { getMigrationTest } from "../migrationUtils";

function convertLegacyConstructorStandingsResponse(constructorStanding: any) {
    const { Constructors, ...rest } = constructorStanding;

    return rest;
}

function convertLegacyResponse(response: any) {
    const { season, round, ConstructorStandings } = response;

    console.log({
        season,
        round,
        constructorStandings: ConstructorStandings.map(convertLegacyConstructorStandingsResponse),
    });

    return { season, round, constructorStandings: ConstructorStandings.map(convertLegacyConstructorStandingsResponse) };
}

describe("GET /constructorStandings smoke tests", () => {
    const migrationTest = getMigrationTest((response) => {
        return response.MRData.StandingsTable.StandingsLists.map(convertLegacyResponse);
    });

    const endpoint = "/constructorStandings";

    const queries: { queryParams: Partial<GetConstructorStandingsDto>; legacyRoute: string }[] = [
        {
            queryParams: { year: 2022 },
            legacyRoute: "2022",
        },
        {
            queryParams: { year: 2009, round: 5 },
            legacyRoute: "2009/5",
        },
        {
            queryParams: { year: 2012, round: 3, constructorId: "ferrari" },
            legacyRoute: "2012/3/constructors/ferrari",
        },
        {
            queryParams: { constructorId: "mercedes" },
            legacyRoute: "constructors/mercedes",
        },
        {
            queryParams: { year: 2019, constructorId: "red_bull" },
            legacyRoute: "2019/constructors/red_bull",
        },
    ];

    migrationTest(endpoint, `https://ergast.com/api/f1/constructorStandings.json`);

    migrationTest(
        `${endpoint}?${querystring.stringify({ position: 2, year: 2022 })}`,
        `https://ergast.com/api/f1/2022/constructorStandings/2.json`
    );

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/constructorStandings.json`
        );
    });
});
