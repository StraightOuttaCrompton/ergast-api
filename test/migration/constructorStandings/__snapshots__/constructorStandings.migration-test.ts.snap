// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /constructorStandings smoke tests /constructorStandings 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "vanwall",
          "name": "Vanwall",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Vanwall",
        },
        "points": "48",
        "position": "1",
        "positionText": "1",
        "wins": "6",
      },
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "40",
        "position": "2",
        "positionText": "2",
        "wins": "2",
      },
      {
        "Constructor": {
          "constructorId": "cooper",
          "name": "Cooper",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "31",
        "position": "3",
        "positionText": "3",
        "wins": "2",
      },
      {
        "Constructor": {
          "constructorId": "brm",
          "name": "BRM",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/BRM",
        },
        "points": "18",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "maserati",
          "name": "Maserati",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Maserati",
        },
        "points": "6",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "team_lotus",
          "name": "Team Lotus",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Team_Lotus",
        },
        "points": "3",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "porsche",
          "name": "Porsche",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Porsche_in_Formula_One",
        },
        "points": "0",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "connaught",
          "name": "Connaught",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Connaught_Engineering",
        },
        "points": "0",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "osca",
          "name": "OSCA",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Officine_Specializate_Costruzione_Automobili",
        },
        "points": "0",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
    ],
    "round": "11",
    "season": "1958",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "cooper-climax",
          "name": "Cooper-Climax",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "40",
        "position": "1",
        "positionText": "1",
        "wins": "5",
      },
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "32",
        "position": "2",
        "positionText": "2",
        "wins": "2",
      },
      {
        "Constructor": {
          "constructorId": "brm",
          "name": "BRM",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/BRM",
        },
        "points": "18",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
      {
        "Constructor": {
          "constructorId": "team_lotus",
          "name": "Team Lotus",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Team_Lotus",
        },
        "points": "5",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "aston_martin",
          "name": "Aston Martin",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Aston_Martin_in_Formula_One",
        },
        "points": "0",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "cooper-maserati",
          "name": "Cooper-Maserati",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "0",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "kurtis_kraft",
          "name": "Kurtis Kraft",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Kurtis_Kraft",
        },
        "points": "0",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "porsche",
          "name": "Porsche",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Porsche_in_Formula_One",
        },
        "points": "0",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "maserati",
          "name": "Maserati",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Maserati",
        },
        "points": "0",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "cooper-borgward",
          "name": "Cooper-Borgward",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "0",
        "position": "10",
        "positionText": "10",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "cooper-osca",
          "name": "Cooper-OSCA",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "0",
        "position": "11",
        "positionText": "11",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "tec-mec",
          "name": "Tec-Mec",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Tec-Mec",
        },
        "points": "0",
        "position": "12",
        "positionText": "12",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "connaught",
          "name": "Connaught",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Connaught_Engineering",
        },
        "points": "0",
        "position": "13",
        "positionText": "13",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "jbw",
          "name": "JBW",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/JBW",
        },
        "points": "0",
        "position": "14",
        "positionText": "14",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "vanwall",
          "name": "Vanwall",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Vanwall",
        },
        "points": "0",
        "position": "15",
        "positionText": "15",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "fry",
          "name": "Fry",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Fry_(racing_team)",
        },
        "points": "0",
        "position": "16",
        "positionText": "16",
        "wins": "0",
      },
    ],
    "round": "9",
    "season": "1959",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "cooper-climax",
          "name": "Cooper-Climax",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "48",
        "position": "1",
        "positionText": "1",
        "wins": "6",
      },
      {
        "Constructor": {
          "constructorId": "team_lotus",
          "name": "Team Lotus",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Team_Lotus",
        },
        "points": "34",
        "position": "2",
        "positionText": "2",
        "wins": "2",
      },
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "26",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
      {
        "Constructor": {
          "constructorId": "brm",
          "name": "BRM",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/BRM",
        },
        "points": "8",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "cooper-maserati",
          "name": "Cooper-Maserati",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Cooper_Car_Company",
        },
        "points": "3",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
    ],
    "round": "10",
    "season": "1960",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?constructorId=mercedes 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "214",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
    ],
    "round": "19",
    "season": "2010",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "165",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
    ],
    "round": "19",
    "season": "2011",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "142",
        "position": "5",
        "positionText": "5",
        "wins": "1",
      },
    ],
    "round": "20",
    "season": "2012",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "360",
        "position": "2",
        "positionText": "2",
        "wins": "3",
      },
    ],
    "round": "19",
    "season": "2013",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "701",
        "position": "1",
        "positionText": "1",
        "wins": "16",
      },
    ],
    "round": "19",
    "season": "2014",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "703",
        "position": "1",
        "positionText": "1",
        "wins": "16",
      },
    ],
    "round": "19",
    "season": "2015",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "765",
        "position": "1",
        "positionText": "1",
        "wins": "19",
      },
    ],
    "round": "21",
    "season": "2016",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "668",
        "position": "1",
        "positionText": "1",
        "wins": "12",
      },
    ],
    "round": "20",
    "season": "2017",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "655",
        "position": "1",
        "positionText": "1",
        "wins": "11",
      },
    ],
    "round": "21",
    "season": "2018",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "739",
        "position": "1",
        "positionText": "1",
        "wins": "15",
      },
    ],
    "round": "21",
    "season": "2019",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "573",
        "position": "1",
        "positionText": "1",
        "wins": "13",
      },
    ],
    "round": "17",
    "season": "2020",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "613.5",
        "position": "1",
        "positionText": "1",
        "wins": "9",
      },
    ],
    "round": "22",
    "season": "2021",
  },
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "515",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
    ],
    "round": "22",
    "season": "2022",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?position=2&year=2022 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "554",
        "position": "2",
        "positionText": "2",
        "wins": "4",
      },
    ],
    "round": "22",
    "season": "2022",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?year=2009&round=5 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "brawn",
          "name": "Brawn",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Brawn_GP",
        },
        "points": "68",
        "position": "1",
        "positionText": "1",
        "wins": "4",
      },
      {
        "Constructor": {
          "constructorId": "red_bull",
          "name": "Red Bull",
          "nationality": "Austrian",
          "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
        },
        "points": "38.5",
        "position": "2",
        "positionText": "2",
        "wins": "1",
      },
      {
        "Constructor": {
          "constructorId": "toyota",
          "name": "Toyota",
          "nationality": "Japanese",
          "url": "http://en.wikipedia.org/wiki/Toyota_Racing",
        },
        "points": "26.5",
        "position": "3",
        "positionText": "3",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "mclaren",
          "name": "McLaren",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/McLaren",
        },
        "points": "13",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "renault",
          "name": "Renault",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Renault_in_Formula_One",
        },
        "points": "9",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "bmw_sauber",
          "name": "BMW Sauber",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/BMW_Sauber",
        },
        "points": "6",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "6",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "williams",
          "name": "Williams",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
        },
        "points": "4.5",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "toro_rosso",
          "name": "Toro Rosso",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Toro_Rosso",
        },
        "points": "4",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "force_india",
          "name": "Force India",
          "nationality": "Indian",
          "url": "http://en.wikipedia.org/wiki/Racing_Point_Force_India",
        },
        "points": "0",
        "position": "10",
        "positionText": "10",
        "wins": "0",
      },
    ],
    "round": "5",
    "season": "2009",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?year=2012&round=3&constructorId=ferrari 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "37",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
    ],
    "round": "3",
    "season": "2012",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?year=2019&constructorId=red_bull 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "red_bull",
          "name": "Red Bull",
          "nationality": "Austrian",
          "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
        },
        "points": "417",
        "position": "3",
        "positionText": "3",
        "wins": "3",
      },
    ],
    "round": "21",
    "season": "2019",
  },
]
`;

exports[`GET /constructorStandings smoke tests /constructorStandings?year=2022 1`] = `
[
  {
    "constructorStandings": [
      {
        "Constructor": {
          "constructorId": "red_bull",
          "name": "Red Bull",
          "nationality": "Austrian",
          "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
        },
        "points": "759",
        "position": "1",
        "positionText": "1",
        "wins": "17",
      },
      {
        "Constructor": {
          "constructorId": "ferrari",
          "name": "Ferrari",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
        },
        "points": "554",
        "position": "2",
        "positionText": "2",
        "wins": "4",
      },
      {
        "Constructor": {
          "constructorId": "mercedes",
          "name": "Mercedes",
          "nationality": "German",
          "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
        },
        "points": "515",
        "position": "3",
        "positionText": "3",
        "wins": "1",
      },
      {
        "Constructor": {
          "constructorId": "alpine",
          "name": "Alpine F1 Team",
          "nationality": "French",
          "url": "http://en.wikipedia.org/wiki/Alpine_F1_Team",
        },
        "points": "173",
        "position": "4",
        "positionText": "4",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "mclaren",
          "name": "McLaren",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/McLaren",
        },
        "points": "159",
        "position": "5",
        "positionText": "5",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "alfa",
          "name": "Alfa Romeo",
          "nationality": "Swiss",
          "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
        },
        "points": "55",
        "position": "6",
        "positionText": "6",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "aston_martin",
          "name": "Aston Martin",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Aston_Martin_in_Formula_One",
        },
        "points": "55",
        "position": "7",
        "positionText": "7",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "haas",
          "name": "Haas F1 Team",
          "nationality": "American",
          "url": "http://en.wikipedia.org/wiki/Haas_F1_Team",
        },
        "points": "37",
        "position": "8",
        "positionText": "8",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "alphatauri",
          "name": "AlphaTauri",
          "nationality": "Italian",
          "url": "http://en.wikipedia.org/wiki/Scuderia_AlphaTauri",
        },
        "points": "35",
        "position": "9",
        "positionText": "9",
        "wins": "0",
      },
      {
        "Constructor": {
          "constructorId": "williams",
          "name": "Williams",
          "nationality": "British",
          "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
        },
        "points": "8",
        "position": "10",
        "positionText": "10",
        "wins": "0",
      },
    ],
    "round": "22",
    "season": "2022",
  },
]
`;
